js/i18n.js

```diff
@@ -345,16 +345,41 @@ function getLanguageListItems($textBox, callback) {
   });
 }
 
+function getMessageString(err, entries, search_for) {
+    var message = "Search error";
+    if (err) {
+        console.log("Error: " + err);
+    } else {
+        if (entries.length == 0) {
+            message = "No matches found for: '" + search_for + "'";
+        } else {
+            var summary = "";
+            var count = 0;
+            entries.forEach(function (entry) {
+                summary += "entry " + (++count) + ": '" + entry.title + "', " + entry.repo_name + "/" + entry.user_name + ", lang=" + entry.lang_code + "\n";
+            });
+            message = count + " Matches found for '" + search_for + "':\n" + summary;
+        }
+    }
+    return message;
+}
+
+function searchAndDisplayResults(search_for) {
+    searchManifest(50, [search_for], null, null, null, null, null,
+        function (err, entries) {
+            var message = getMessageString(err, entries, search_for);
+            alert(message);
+        }
+    );
+}
 
 $().ready(function () {
   window.setTimeout(searchForResources(''), 300);
   window.setTimeout(setupLanguageSelector(), 500);
 
   $('#search-td').on('click', function (){
     var search_for = document.getElementById('search-for').value;
```

Comment:
While you didn't write this line, can we get all code being jQuery if that is what we are using? Should be `var search_for = $('#search-for').val();`